### DISCUSSION TITLE
Having issue setting up the connection with oracle db using oracledb

### DISCUSSION THREAD

**Discussion by jyonag** [oracle/python-oracledb#305] [Q&A]
Tried to install oracledb in my VM but couldnt do it because of the network firewalls. I tied to install install them locally and copied the file and folders to python "C:\Program Files (x86)\Python38-32\site-packages" 
![image](https://github.com/oracle/python-oracledb/assets/152456571/45321dc3-5f65-43ff-b27f-f96091e1ba50)
I created sample program to test the db connection test.py
![image](https://github.com/oracle/python-oracledb/assets/152456571/7bd3770e-a5a3-4a77-85df-d5b9ec2215bc)
When I tried to test the connection. I got error as follows
![image](https://github.com/oracle/python-oracledb/assets/152456571/5b0081ff-10e9-4a3f-ae1c-812b0cc93d08)
C:\Program Files (x86)\Python38-32\site-packages>python test.py
Traceback (most recent call last):
  File "test.py", line 3, in <module>
    import oracledb
  File "C:\Program Files (x86)\Python38-32\site-packages\oracledb\__init__.py", line 169, in <module>
    from .connection import (
  File "C:\Program Files (x86)\Python38-32\site-packages\oracledb\connection.py", line 45, in <module>
    from . import base_impl, thick_impl, thin_impl
ImportError: cannot import name 'base_impl' from partially initialized module 'oracledb' (most likely due to a circular import) (C:\Program Files (x86)\Python38-32\site-packages\oracledb\__init__.py)

**Comment by cjbj** [comment]
There were changes made in https://github.com/oracle/python-oracledb/commit/891e3fa7ed450602481372f1c5c4e2d96cb3b3cc to address this issue. You now might find it easier to resolve any base issue.
